### DISCUSSION TITLE
How does Philadelphia compare to commercial Java low latency FIX engines?

### DISCUSSION THREAD

**Discussion by Artie314** [paritytrading/philadelphia#230] [General]
Hi, I am evaluating Java FIX engines now for my next project and wondering if anyone knows how Philadelphia compares to some of the commercial Java low latency FIX engines like Chronicle or CoralFIX? Both in performance and in functionality. Any advice would be appreciated. Thanks!

**Comment by jvirtanen** [comment]
Why the removal? 🤔  I think this (how Philadelphia compares to commercial low-latency FIX engines, such as Chronicle FIX and CoralFIX) is a genuinely interesting topic. It's just that no one has got answering the question yet. (A reminder: as far as I know, no one gets paid working on this project.)

**Reply by Artie314** [reply]
After further analysis I do not think Philadelphia will suit my requirements so was no longer interested. But I agree it is an interesting topic. Putting aside the performance comparison, does Philadelphia support some of the general FIX engine features such as FIX message validation, scheduling of FIX sessions, auto reconnect, sequence number management, etc? Or are these things typically built on top of the library when needed? Is there any documentation on what features are currently supported? Thanks!

**Reply by jvirtanen** [reply]
Hi @Artie314 and my apologies for a slow response! 👋🏼 

Philadelphia does not support any of the aforementioned features out of the box, so these have traditionally been built on top of it if needed. One of the objectives in the near future is to make it possible for applications to implement their own handling for all administrative messages, which should help with this.

**Comment by jvirtanen** [comment]
Hi @Artie314! Unfortunately I don't have any first-hand experience from either Chronicle FIX or CoralFIX. However, in 2020 I heard from a person who was evaluating a few options in the space that with some small tweaks, Philadelphia performed favorably when compared to Chronicle FIX. However, when it comes to features, I think both Chronicle FIX and CoralFIX are quite a bit more extensive than Philadelphia.

**Comment by Artie314** [comment]
Thank you for your reply!
